### PR TITLE
[Spanner] Add timeout + retry settings to Sessions/Snapshots

### PR DESCRIPTION
**Pull request by crwilcox**

### CODE REVIEW

File: spanner/google/cloud/spanner_v1/gapic/spanner_client_config.py [hunk 1]
```diff
@@ -16,6 +16,15 @@
...
                 },
+                "streaming": {
+                    "initial_retry_delay_millis": 1000,
+                    "retry_delay_multiplier": 1.3,
+                    "max_retry_delay_millis": 32000,
+                    "initial_rpc_timeout_millis": 300000,
+                    "rpc_timeout_multiplier": 1.0,
+                    "max_rpc_timeout_millis": 300000,
+                    "total_timeout_millis": 3000000
+                },
```
Comment (tseaver) [inline spanner/google/cloud/spanner_v1/gapic/spanner_client_config.py]: Is this manual hackage?  I don't see recent changes to the `googleapis` repository for spanner.

File: spanner/tests/unit/test_session.py [hunk 2]
```diff
@@ -397,6 +399,8 @@ def test_execute_sql_explicit(self):
             params,
             param_types,
             'PLAN',
+            timeout=google.api_core.gapic_v1.method.DEFAULT,
+            retry=google.api_core.gapic_v1.method.DEFAULT,
```
Comment (tseaver) [inline spanner/tests/unit/test_session.py]: The `_explicit` version of the test should pass / assert non-default values for `timeout` / `retry`.

File: spanner/tests/unit/test_snapshot.py [hunk 3]
```diff
@@ -417,6 +417,8 @@ def _execute_sql_helper(
             partition_token=partition,
             seqno=sql_count,
             metadata=[('google-cloud-resource-prefix', database.name)],
+            timeout=google.api_core.gapic_v1.method.DEFAULT,
+            retry=google.api_core.gapic_v1.method.DEFAULT
```
Comment (tseaver) [inline spanner/tests/unit/test_snapshot.py]: This helper should probably grow support for passing explicit `timeout` / `retry` arguments;  we would then need a driver testcase (or two) which exercises that.

File: spanner/google/cloud/spanner_v1/gapic/spanner_client_config.py (hunk 1 above)
Comment (crwilcox) [reply spanner/google/cloud/spanner_v1/gapic/spanner_client_config.py]: Correct.

Comment (tseaver) [reply spanner/google/cloud/spanner_v1/gapic/spanner_client_config.py]: Shouldn't this be done via `synth.py`, then?

Comment (crwilcox) [reply spanner/google/cloud/spanner_v1/gapic/spanner_client_config.py]: Making this change on our generated side. will strip from this pr once that is in.

Comment (tseaver) [reply spanner/google/cloud/spanner_v1/gapic/spanner_client_config.py]: 779569ac758dd0a789f4ed8c84c7656c28ca1de3

File: spanner/tests/unit/test_session.py (hunk 2 above)
Comment (tseaver) [reply spanner/tests/unit/test_session.py]: 80e22369b5fe1d5ddb6a9f00a0663e5acd7ac6f2

File: spanner/tests/unit/test_snapshot.py (hunk 3 above)
Comment (tseaver) [reply spanner/tests/unit/test_snapshot.py]: 80e22369b5fe1d5ddb6a9f00a0663e5acd7ac6f2

**Pull request merged**